pkg/scalers/datadog_scaler.go

```diff
@@ -0,0 +1,316 @@
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("queryValue parsing error %s", err.Error())
+		}
+		meta.queryValue = queryValue
+	} else {
+		return nil, fmt.Errorf("no queryValue given")
+	}
+
+	if val, ok := config.TriggerMetadata["age"]; ok {
+		age, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("age parsing error %s", err.Error())
+		}
+		meta.age = age
+	} else {
+		meta.age = 90 // Default window 90 seconds
+	}
+
+	// For all the points in a given window, we take the rollup to the window size
+	rollup := fmt.Sprintf(".rollup(avg, %d)", meta.age)
+	meta.query = meta.query + rollup
+
+	if val, ok := config.TriggerMetadata["type"]; ok {
+		val = strings.ToLower(val)
+		switch val {
+		case "average":
+			meta.vType = average
+		case "global":
+			meta.vType = global
+		default:
+			return nil, fmt.Errorf("type has to be global or average")
+		}
+	} else {
+		meta.vType = average // Default to average between pods
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
+		meta.metricName = GenerateMetricNameWithIndex(config.ScalerIndex, kedautil.NormalizeString(fmt.Sprintf("datadog-%s", val)))
+	} else {
+		return nil, fmt.Errorf("no metric name given")
+	}
+
+	if val, ok := config.AuthParams["apiKey"]; ok {
+		meta.apiKey = val
+	} else {
+		return nil, fmt.Errorf("no api key given")
+	}
+
+	if val, ok := config.AuthParams["appKey"]; ok {
+		meta.appKey = val
+	} else {
+		return nil, fmt.Errorf("no app key given")
+	}
+
+	if val, ok := config.AuthParams["datadogSite"]; ok {
+		if val != "" {
+			meta.datadogSite = val
+		} else {
+			meta.datadogSite = "datadoghq.com"
+		}
+	} else {
+		meta.datadogSite = "datadoghq.com"
+	}
+
+	return &meta, nil
+}
+
+// newDatddogConnection tests a connection to the Datadog API
+func newDatadogConnection(ctx context.Context, meta *datadogMetadata) (*datadog.APIClient, error) {
+
+	ctx = context.WithValue(
+		context.Background(),
+		datadog.ContextAPIKeys,
+		map[string]datadog.APIKey{
+			"apiKeyAuth": {
+				Key: meta.apiKey,
+			},
+			"appKeyAuth": {
+				Key: meta.appKey,
+			},
+		},
+	)
+
+	ctx = context.WithValue(ctx,
+		datadog.ContextServerVariables,
+		map[string]string{
+			"site": meta.datadogSite,
+		})
+
+	configuration := datadog.NewConfiguration()
+	apiClient := datadog.NewAPIClient(configuration)
+
+	_, _, err := apiClient.AuthenticationApi.Validate(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to Datadog API endpoint: %v", err)
+	}
+
+	return apiClient, nil
+}
+
+// No need to close connections
+func (s *datadogScaler) Close(context.Context) error {
+	return nil
+}
+
+// IsActive returns true if we are able to get metrics from Datadog
+func (s *datadogScaler) IsActive(ctx context.Context) (bool, error) {
+
+	ctx = context.WithValue(
+		context.Background(),
```

Comment:
similar to the above, why not use the given `ctx` parameter here?

```suggestion
		ctx,
```